portfolio/src/main/webapp/visuals.js

```diff
@@ -0,0 +1,255 @@
+    for (const i in cuisineVotes) {
+      const cuisineName = cuisineVotes[i].cuisineName;
+      const votes = cuisineVotes[i].votes;
+      let element = document.createElement("option");
+      const text = document.createTextNode(cuisineName);
+      element.appendChild(text);
+      element.setAttribute("value", cuisineName);
+      document.getElementById("cuisine-select").appendChild(element);
+
+      if (votes > 0) {
+        data.addRow([cuisineName, votes]);
+      }
+    }
+
+    const options = {
+      pieHole: 0.4,
+      pieSliceTextStyle: {
+        color: '#f5f6fa'
+      },
+      backgroundColor: '#a6ded2',
+      legend: 'none'
+    };
+
+    const chart = new google.visualization.PieChart(document.getElementById('chart-container'));
+    chart.draw(data, options);
+  })
+}
+
+/**
+ * Adds the user's vote to the database and updates the chart accordingly. If the
+ * user has already voted then their vote is just updated (this way they cannot vote twice).
+ */
+function addCuisineVote() {
+  const selectVal = document.getElementById("cuisine-select").value;
+  if (selectVal !== "") {
+    let userId = getCookie("userId");
+    let cuisineVote = getCookie("cuisineVote");
+    if (cuisineVote !== selectVal) {
+      if (userId === "") {
+        userId = Date.now().toString();
+        document.cookie = 'userId=' + userId;
+      }
+      cuisineVote = selectVal;
+      document.cookie = 'cuisineVote=' + selectVal;
+
+      let cuisineVoteJson = {
+          'userId': userId,
+          'cuisine': cuisineVote
+      };
+
+      fetch('/cuisine-data', {method: 'POST',
+        headers: {'Content-Type': 'application/json'},
+        body: JSON.stringify(cuisineVoteJson)
+        }).then(response => response.text()).then((data) => {
+          drawChart();
+        })
+    }
+  }
+}
+
+/** Returns the value mapped to the inputted parameter. */
+function getCookie(name) {
+  let key = name + "=";
+  let decodedCookie = decodeURIComponent(document.cookie);
+  let cookies = decodedCookie.split(';');
+  for(let i = 0; i < cookies.length; i++) {
+    let cookie = cookies[i];
+    cookie = cookie.trim();
+    if (cookie.indexOf(key) == 0) {
+      return cookie.substring(key.length);
+    }
+  }
+  return "";
+}
+
+/** Creates a map and adds it to the page. */
+function createMap() {
+  fetch('/basketball-courts').then(response => response.json()).then((courts) => {
+    map = new google.maps.Map(document.getElementById('map-container'), {
+      center: { lat: 40.7128, lng: -74.0060 },
+      zoom: 10,
+      styles: style
+    });
+
+    courtsInfo = courts;
+  });
+}
+
+/**
+ * Add markers for courts.
+ * The "i * 200" essentially means that each marker will be placed 200ms
+ * after each other. It's a good value because it is short enough that the
+ * user won't be waiting too long for the pins to drop, but also long enough that
+ * the user can see the effect of the pins dropping individually.
+ */
+function showCourtsMarkers() {
```

Comment:
wondering why cuisine and basketball courts logic and visuals are all in the same visuals.js file